pxtcompiler/emitter/transpile.ts

```diff
@@ -0,0 +1,115 @@
+// TODO: enable reference so we don't need to use: (pxt as any).py
+//      the issue is that this creates a circular dependency. This
+//      is easily handled if we used proper TS modules.
+//// <reference path="../../built/pxtpy.d.ts"/>
+
+// This code centralizes transpiling between typescript and python
+// so that we can cache results for better performance and user
+// experience (since compile -> decompile round-trips are lossy)
+namespace ts.pxtc.transpile {
+    export interface TranspileResult {
+        diagnostics: pxtc.KsDiagnostic[],
+        success: boolean,
+        outfiles: { [key: string]: string }
+    }
+
+    const mainName = (l: Lang) => `main.${l}`
```

Comment:
Just a function?

Reply:
sure, but this is one line. It's a pattern we use liberally elsewhere.

![image](https://user-images.githubusercontent.com/6453828/69826450-0730ee00-11c8-11ea-9617-b2ad2a867fc1.png)